transformers/modeling_seq2seq.py

```diff
@@ -108,23 +113,28 @@ def from_pretrained(cls, encoder_pretrained_model_name_or_path=None, decoder_pre
 
         # Separate the encoder- and decoder- specific kwargs. A kwarg is
         # decoder-specific it the key starts with `decoder_`
-        kwargs_decoder = {}
-        kwargs_encoder = kwargs
-        for key in kwargs_encoder.keys():
-            if key.startswith("decoder_"):
-                kwargs_decoder[key.replace("decoder_", "")] = kwargs_encoder.pop(key)
+        kwargs_encoder = {
```

Comment:
Here we should have three possibilities:
- arguments without an `encoder_` or `decoder_` prefix should be sent to both (e.g. a `cache_dir` argument).
- arguments with `encoder_` should be sent to the encoder only
- arguments with `decoder_` should be sent to the decoder only.

Can you update this (the first condition in particular)?

Reply:
Yes, it is much better this way. I also made sure encoder- or decoder-specific arguments override the "common" ones if there is a conflict.